Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.cpp

```diff
@@ -131,33 +131,47 @@ namespace AZ
             // the child material. See https://github.com/o3de/o3de/issues/13766
             if (!materialTypePath.empty())
             {
-                // If the material uses the "Direct" format, then there will need to be a dependency on that file.
-                // If it uses the "Abstract" format, then there will be an intermediate .materialtype and there needs to be a dependency on that file instead.
-                // At this point the builder does not know which is the case, without loading the .materialtype file and inspecting its data. The builder
-                // avoids that because it could slow things down, and instead just registers both dependencies.
+                // Load the material type data to find the exact material type format.
+                // This is required to get an accurate dependency
+                MaterialTypeSourceData::Format materialTypeForamt = MaterialTypeSourceData::Format::Invalid;
+                MaterialUtils::ImportedJsonFiles importedJsonFiles;
+                auto materialTypeSourceData = MaterialUtils::LoadMaterialTypeSourceData(materialTypePath, nullptr, &importedJsonFiles);
```

Comment:
If this function opens a file maybe add a comment about why this is not the normal behavior in CreateJobs but we are making an exception + reasons for the exception